wordpress-mobile/WordPress-Android
# Fix n-dash in translations

Our lint rules include checking that `n-dash` (and not standard or `m-`) is used as a separator in numbers. 
Lately, we've had some issues with this because many translators don't notice the difference and GlotPress doesn't offer any support for this. In many cases, the lint checks failed on new translations and we had to manually fix them (eg: https://github.com/wordpress-mobile/WordPress-Android/commit/2e693352fb5aef19277768c3a6a0cff34e9c835b) 

This PR adds a new regex to the sed pipe used to format the strings during the download which replaces the standard dash with `n-dash`. 

_The regex_
This is the used regex: `/Translation-Revision-Date/!s/([[:digit:]])-([[:digit:]])/\1–\2/g`:
- `/([[:digit:]])-([[:digit:]])/` matches a _digit -> dash -> digit_ sequence and capture the digit before and the one after the dash. 
- `\1–\2` outputs the first digit, then the `n-dash`, then the second digit. 
- `/Translation-Revision-Date/!`: this is a hacky solution to avoid the dash replacement in the dates in the comment at the top of the file. The linter is happy either ways and I don't think the change makes sense there. 


_To test_:
1. Verify that CI is green (I've downloaded the new strings with the new script and committed them as a part of this PR).
2. Run `./tools/update-translations.sh` to download the latest strings and run `./gradlew lintVanillaRelease` (or any other flavour) to verify that the `n-dash` is used when required. 

**Note:** I've opened this PR against the release branch because it affects only the tooling and I think that it'll be handy to have for the next release, but I'm totally ok with moving it to target `develop` if you prefer it. 

PR submission checklist:
